File: sorting/bubble.go
package main

// BubbleSort ...
func BubbleSort(arr []int) {
	for i := range arr {
		for j := range arr[:len(arr)-i-1] {
			if arr[j] > arr[j+1] {
				arr[j], arr[j+1] = arr[j+1], arr[j]
			}
		}
	}
}

File: sorting/bubble.js
const swap = (arr, i, j) => {
    let tmp = arr[i]
    arr[i] = arr[j]
    arr[j] = tmp
}

function bubbleSort(arr) {
    for (let i = 0; i < arr.length; i++) {
        for (let j = 0; j < arr.length - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                let tmp = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = tmp
            }
        }
    }
}

const arr = [3, 4, 1, 5, 6, 1, 7]
bubbleSort(arr)
console.log(arr)